libs/shared/storage/src/lib/shared-storage.service.ts

```diff
@@ -126,19 +127,24 @@ export class SharedStorageService {
   }
 
   public async putObject(
-    id: string,
+    key: string,
     data: Buffer | Readable,
     isPrivate = false,
   ): Promise<AWS.S3.ManagedUpload.SendData> {
     const acl = isPrivate ? 'private' : 'public-read';
     const request: AWS.S3.PutObjectRequest = {
-      Key: id,
-      Body: data,
+      Key: key,
+      Body: isReadableStream(data) 
```

Comment:
Doesn't this set the body to be a boolean?

Reply:
The value of the body key continue onto the next two lines:
```
Body: isReadableStream(data) 
        ? new PassThrough() 
        : data,
```